include/wx/cmndata.h

```diff
@@ -222,6 +244,9 @@ class WXDLLIMPEXP_CORE wxPrintDialogData: public wxObject
     bool            m_printEnablePrintToFile = true;
     wxPrintData     m_printData;
 
+    int m_maxPageRanges = 64;
```

Comment:
It would be nice to have a comment here explaining where does this value come from and, more generally, why do we need this variable (and the associated functions) at all.

Reply:
with m_maxPageRanges  you can set the maximum number of ranges allowed. This values is used for the internal array that is required to hold the ranges. When the print dialog is opened, the user can then define up to m_maxPageRanges  ranges because the internal array can only hold this amount of ranges.

Reply:
Ah, I see, thanks for the explanation.

But I believe that the default value of 64 should really be fine, at least I can't imagine anybody needing more ranges than this in any kind of realistic scenario. So my first instinct would be to just leave this field and these constants off and use 64 in wxMSW code.

Of course, if you really want to have them (in order to increase this even further, I guess?), we can keep them, but in this case we should explain in the docs that `SetMaxPageRanges()` can be called to increase the default limit and mention what it is. So it would be just simpler to not do this and not have all this at all.

Reply:
I think that eventually someone might want to make the number of regions smaller rather than larger.

Reply:
This is what I don't understand: why would anybody want to make this smaller or restrict it at all? I.e. I understand that we need to specify some number of ranges in advance because this is how Win32 API works and I can accept that, in theory, any number we specify might be too small for somebody (even though in practice 64 should be more than enough), but what would be the motivation for restricting the number of ranges?

I think the only cases we need handle are:

1. Not accepting ranges at all.
2. Accepting exactly one range.
3. Accepting any number of them.

Am I missing something?

Reply:
Yes, and `SetMaxPageRanges`  covers all three cases you mentioned and additionally a limit number of ranges, depending on the app that uses this code. 

I think we shouldn't limit the space of possibilities beforehand just because we can't currently imagine that there are other cases besides the 3 cases that you mentioned.

I would therefore strongly recommend letting the developer decide how many page ranges the user of an app can enter.

Reply:
Sorry, I disagree here. We shouldn't be adding API to do something that we have no realistic use case for. It can always be added later if needed, but removing it is much harder.

So if there is really no concrete scenario in which these functions are useful, let's remove them for now.

Reply:
You mentioned to add the method `SetEnableMultiplePageRanges` to limit the number of ranges to one. I would not do that, I would use `SetMaxPageRanges` because SetEnableMultiplePageRanges is just a sub-case for SetMaxPageRanges. 

We could even limit the default number of ranges to one and then developer can increase this number with  `SetMaxPageRanges` to allow multiple ranges.  So the developer has more flexibility if we have `SetMaxPageRanges`.

Reply:
The problem is that using `SetMaxPageRanges()` requires the application programmer to arbitrarily select the maximum number of supported ranges. This makes the API more difficult to use and increases the possibility of someone making a wrong choice and passing 10,000 or something like that to it, which would certainly be a bad idea. I'd like to avoid people having to make this arbitrary choice and just allowing them to indicate that they want (and are ready) to support multiple page ranges, without specifying how many of them exactly do they need because there is just no good answer to this question in general.